### PR TITLE
🤬[Techdebt] #167 add custom user agent++

### LINKED ISSUE #156: 🦺[Tech Debt] Reduce Cocoapods and Increase Swift Packages
## Goal 
**Describe the work to be done**
 
Check to see where current Cocoapods cn be converted to Swift Packages

Current Podfile:
```
  pod 'UnstoppableDomainsResolution', '~> 4.0.0'
  pod 'KeychainAccess', '~> 4.2'
  pod 'Firebase/Analytics'
  pod 'Firebase/Crashlytics' 
```

## Definition of Done
- [ ] Do this incrementally  
- [ ] Update the Bitrise CI/CD Step to remove the Podfile

### LINKED ISSUE #167: 🦺[Tech Debt] Add Custom User Agent
## Goal 
Added a custom user agent string to aid business goals

## Definition of Done
- [ ] Adde custom user agent for all WKWebView
- [ ] String: `litewallet-ios`

### LINKED ISSUE #170: 🥳[Feature]  Replace PartnerAPI with WebView for Litecoin Card
## Overview
The API integration of Litecoin Card never worked as intended and causes problems for users.
We also have to maintain a lib where are partner are not supporting features.
The best scenario is to replace the integration with a webview.
### Why?
**Is your request related to a problem? Or, do you have an idea to improve the app? Please describe.**
 
A few good reasons to do this:
- Ternio/Unbanked are not supporting the integration
- We can direct users to the actual site
- No longer do we have to carry the library
 
### Definition of Done
- [ ] Replace view with WKWebView to `https://litecoin.dashboard.getblockcard.com/`
- [ ] Remove LitewalletPartnerAPI

### PR DESCRIPTION
**Pull request by kcw-grunt**
# Overview
Like many things in software development, this started as a simple fix/update.  But, it became much more....

## TL; DR

This PR includes the following fixes:

-  #167 (Adding a small customization for sales tracking)
- Rebasing from the `main` branch
- Rebasing from the `develop` branch
- Refactor to make the automatic binary deliver to TestFlight
- **Removal of Cocoapods** (SPM and Git Submodules only)
- Addition of  the private repo `LitewalletPartnerAPI` via Git Submodule
- Replace integrated LitecoinCard  code with WebView

## Participate in [Schadenfreude](https://en.wikipedia.org/wiki/Schadenfreude)!

Walk with me in this journey....have a laugh.

As mentioned, the first effort was to add a small 0.5 point fix that was really done in 10 mins.  If you see references to `customUserAgent` this it the extent of that work.

After that was done the effect to actually see if the TestFlight distribution was the other effort and it was clear there were still issues with the Bitrise CI/CD workflow.  Its wonderful that we have been able to successfully deliver the .ipa to TestFlight, the workflow for PRs is still broken.

Other correspondence with Bitrise proved that, in Bitrise, has a bug in their system.  They admitted as such:

| Admission of Guilt by Bitrise 1 | Admission of Guilt by Bitrise 2 |
|--------------------------------|-----------------------------|
|Screen Shot 2022-12-03 at 7 20 28 PM|Screen Shot 2022-12-07 at 11 30 20 AM|

# Closes #167 
Simply at a sting for sales tracking

# Closes #156
Throughout the effort, the opportunity to refactor the Cocoapods to Swift Package presented itself and the action was taken. This closed #156 

# Closes #170 
But, by converting to SPM, the partnerAPI package broke the Bitrise CI process. SPM doesn't handle private packages very well.  But, truth is the Litecoin Card integration never worked well, so the initial work was needed when the Card was launched.  Today, the partner (Ternio/Unbanked) has a full-featured web presence and replacing the View with a simple webview.  So, this work also closes #170 

Finally the re-base of the develop and main branch meant many .ymls efforts shows many adjustments in the numerous commits ...very sorry about this.

|New Look Litecoin Card WebView|
|---------------------------------|
|webview-card|

### CODE REVIEW

**Comment by ios-agent-litewallet** [comment]
SUCCESSFUL-BUILD-NUMBER-1479

File: loafwallet/src/Constants/Constants.swift [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import UIKit
 
 let π: CGFloat = .pi
+let customUserAgent: String = "litewallet-ios"
```
Comment (kcw-grunt) [inline loafwallet/src/Constants/Constants.swift:4]: The source of the small fix to use a reference for sales tracking

**Comment by kcw-grunt** [comment]
CURRENT-TEST-SUCCESS-1584

**Comment by kcw-grunt** [comment]
CURRENT-TEST-SUCCESS-1586

**Comment by kcw-grunt** [comment]
CURRENT-TEST-SUCCESS-1587

**Comment by kcw-grunt** [comment]
CURRENT-TEST-SUCCESS-1589

**Comment by kcw-grunt** [comment]
CURRENT-TEST-SUCCESS-1590

**Comment by kcw-grunt** [comment]
CURRENT-TEST-SUCCESS-1592

File: Podfile [hunk 2]
```diff
@@ -1,39 +0,0 @@
...
-  
-  #Sets all pods to iOS 10.0 or greater
-  installer.pods_project.targets.each do |target|
-    target.build_configurations.each do |config|
-      config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '10.0'
-    end
-  end
-  
-end
-
```
Comment (Iferencak) [inline Podfile:39]: Yaaaaay 🎉

File: loafwallet/ForgotAlertViewModel.swift [hunk 3]
```diff
@@ -13,13 +12,13 @@ class ForgotAlertViewModel: ObservableObject {
...
-				completion()
-			}
-		}
+	func resetPassword(completion _: @escaping () -> Void) {
+//		PartnerAPI.shared.forgotPassword(email: emailString) { responseMessage in
+//			DispatchQueue.main.async {
+//				self.detailMessage = responseMessage
+//				completion()
+//			}
+//		}
```
Comment (Iferencak) [inline loafwallet/ForgotAlertViewModel.swift]: Is this not used anymore? If so maybe remove the whole function? But It looks like it should be used?

File: loafwallet/src/Strings/de.lproj/Localizable.strings [hunk 4]
```diff
@@ -1446,3 +1443,6 @@
 
 /* 1 of 3 */
 "WritePaperPhrase.step" = "%1$d von %2$d";
+
+/* UDSystemError */
+"Send.UnstoppableDomains.udSystemError" = "";
```
Comment (Iferencak) [inline loafwallet/src/Strings/de.lproj/Localizable.strings]: Is this error no longer neccessary? If so, maybe it can be removed?

File: loafwallet/src/Strings/es.lproj/Localizable.strings [hunk 5]
```diff
@@ -1448,3 +1445,6 @@
 
 /* 1 of 3 */
 "WritePaperPhrase.step" = "%1$d de %2$d";
+
+/* UDSystemError */
+"Send.UnstoppableDomains.udSystemError" = "";
```
Comment (Iferencak) [inline loafwallet/src/Strings/es.lproj/Localizable.strings]: Same comment on all translations.

File: loafwallet/TransferAmountViewModel.swift [hunk 6]
```diff
@@ -89,16 +89,17 @@ class TransferAmountViewModel: ObservableObject {
...
+		//			.withdrawToWallet(userID: userID,
+		//			                  token: token,
+		//			                  withdrawal:
+		//			                  ["amount": amount,
+		//			                   "wallet_address": address]) { _ in
+		//
+		//			}
 	}
 
 	/// Transfer Litecoin from **Litewallet to Litecoin Card**
```
Comment (Iferencak) [inline loafwallet/TransferAmountViewModel.swift:95]: As TransferAmountView was deleted, seems like we don't need this viewmodel anymore?

File: loafwallet/CardWebViewController.swift [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+	}
+}
+
+extension CardWebViewController: WKNavigationDelegate {
+	func webView(_: WKWebView, didCommit _: WKNavigation!) {}
+
+	func webView(_: WKWebView, didFinish _: WKNavigation!) {}
+
+	func webView(_: WKWebView, didFail _: WKNavigation!, withError _: Error) {}
+}
```
Comment (Iferencak) [inline loafwallet/CardWebViewController.swift]: This all seems like unused code, maybe we should just remove it?

File: loafwallet/ForgotAlertViewModel.swift (hunk 3 above)
Comment (kcw-grunt) [reply loafwallet/ForgotAlertViewModel.swift]: Yeah I think there is alot of dead code now related to the PartnerAPI...removing

File: loafwallet/src/Strings/de.lproj/Localizable.strings (hunk 4 above)
Comment (kcw-grunt) [reply loafwallet/src/Strings/de.lproj/Localizable.strings]: This is a merge conflict problem I created...re-adding per the base localization

File: loafwallet/CardWebViewController.swift (hunk 7 above)
Comment (kcw-grunt) [reply loafwallet/CardWebViewController.swift]: My mistake. Thanks!

**Comment by kcw-grunt** [comment]
CURRENT-TEST-SUCCESS-1594

**Comment by kcw-grunt** [comment]
Good typical code review  even when i'm being sloppy..😅
Very nice @Iferencak ...thanks so much.
cc: @vsima @remihoh

**Comment by kcw-grunt** [comment]
When you have a chance @Iferencak
